nipoppy/workflows/base.py

```diff
@@ -297,14 +297,20 @@ def __init__(
         _validate_layout : bool, optional
             If True, validate the layout during setup, by default True
         """
-        super().__init__(name=name, verbose=verbose, dry_run=dry_run)
-
-        self.dpath_root = Path(dpath_root)
+        self.dpath_root = is_nipoppy_project(dpath_root)
+        # `.nipoppy` is not created by default in version 0.3.4 and below
+        if self.dpath_root is False:
+            self.dpath_root = Path(dpath_root)
```

Comment:
Can this be a one-liner instead? Or is that not a good idea (is `pathlib.Path` always truthy?)
```suggestion
        # `.nipoppy` is not created by default in version 0.3.4 and below
        self.dpath_root = is_nipoppy_project(dpath_root) or Path(dpath_root)
```